quinn-proto/src/packet.rs

```diff
@@ -169,6 +212,35 @@ impl PartialDecode {
             if allow_coalesced { Some(bytes) } else { None },
         ))
     }
+
+    fn get_packet_number(
+        buf: &mut io::Cursor<BytesMut>,
+        pn_key: &PacketNumberKey,
+        sample_offset: usize,
+    ) -> Result<PacketNumber, PacketDecodeError> {
+        let mut first = [buf.bytes()[0]; 1];
+        let sample = {
+            let mut sample = [0; 16];
+            debug_assert!(pn_key.sample_size() <= 16);
+            sample.copy_from_slice(
+                &buf.get_ref()[sample_offset..sample_offset + pn_key.sample_size()],
```

Comment:
How do we know this isn't out of bounds?

Reply:
I've added a bounds check.